sys/fuchsia/fidl_process.txt

```diff
@@ -0,0 +1,28 @@
+include <zircon/syscalls.h>
+include <zircon/types.h>
+
+# launcher.fidl
+
+launch_info {
+	executable	zx_vmo
+	job		zx_job
+	name		fidl_string
+} [align_8]
+
+launch_result {
+	status		int32
+	error_message	fidl_string
+	process		zx_process
+} [align_8]
+
+fuchsia_process_LauncherLaunchRequest {
+	hdr	fidl_message_header
+	info	launch_info
+} [align_8]
+
+fuchsia_process_LauncherLaunchResponse {
+	hdr	fidl_message_header
+	result	launch_result
+} [align_8]
+
+syz_fidl_send_msg$fuchsia_process_LauncherLaunch(handle zx_chan, type ptr[in, string["fuchsia.process.Launcher"]], ordinal const[1], request ptr[in, fuchsia_process_LauncherLaunchRequest], response ptr[out, fuchsia_process_LauncherLaunchResponse])
```

Comment:
If we figure out how to make syzkaller generate already serialized messages, then we can replace this with just zx_channel_call specialization.

And for output arguments we never care about actual contents, right? If the service call returns any handles, then they will be in zx_channel_call_args_t.rd_handles, and we don't care about anything else it can return. So we could replace the output argument with just:
```
define process_LauncherLaunchResponse_size sizeof(fuchsia_process_LauncherLaunchResponseTable)
```
and then use `array[int8, process_LauncherLaunchResponse_size]`.

Reply:
Regarding serialization of request (and deserialization of response), I will follow-up with another patch. For now, I just added a zx_channel_call specialization that does not actually specialize its arguments.